draft-steele-cose-merkle-tree-proofs.md

```diff
@@ -89,51 +89,50 @@ This document describes the three data structures necessary to use merkle proofs
 
 # Terminology
 
-Leaf Bytes:
+Leaf:
 
-: A merkle tree leaf is labelled with the cryptographic hash of a sequence of bytes.
-These bytes may be structured as a combination of Payload and Extra Data.
+: A merkle tree leaf is the cryptographic hash of a sequence of bytes that combines Payload and Extra Data.
 
 Merkle Tree:
 
-: A Merkle tree is a tree where every leaf is labelled with the cryptographic hash of a sequence of
-bytes and every node that is not a leaf is labeled with the cryptographic hash of the labels of its child nodes.
+: A Merkle tree is a tree where every leaf is a cryptographic hash of a sequence of
+bytes and every node that is not a leaf is the cryptographic hash of the its child nodes.
 
-Merkle Tree Root:
+Merkle Root:
 
-: A Merkle tree root is the root node of a tree which represents the cryptographic hash that commits to all leaves in the tree.
+: A Merkle root is the root node of a tree; this cryptographic hash is a committment to the content of the tree.
 
 Merkle Tree Algorithm:
 
-: A Merkle tree algorithm specifies how nodes in the tree must be hashed to compute the root node.
+: A Merkle tree algorithm specifies how to construct the tree and how to compute its root.
 
 Payload and Extra Data:
 
-: A payload is data bound to in a Merkle tree leaf.
-The Merkle tree algorithm determines how a payload together with extra data is bound to a leaf.
-The simplest case is that the payload is the leaf itself without extra data.
+: A payload is application data used to produce a Leaf.
+The Merkle tree algorithm determines how a payload together with extra data is used to produce a leaf.
+The simplest case is that the leaf is the cryptographic hash of the payload without extra data.
 
 Inclusion Path:
 
-: An inclusion path confirms that a value is a leaf of a Merkle tree known only by its root hash (and tree size, possibly).
+: The inclusion path enables a verifier to recompute a root, given a leaf.
 
-Signed Merkle Tree Proof:
+Signed Inclusion Proof:
 
-: A signed Merkle tree proof is the combination of signed Merkle tree root hash, inclusion path, extra data, and payload.
+: A signed inclusion proof is a combination of payload, extra data, inclusion path and signed envelope that includes a merkle root.
```

Comment:
We can do that but I think we should also define something that doesn't assume a signed envelope, otherwise it wouldn't be general enough.

Reply:
I added this in https://github.com/ietf-scitt/draft-steele-cose-merkle-tree-proofs/pull/9/commits/49c6f38c05e7882445f395acf977d9cad8d40c78